content/en/docs/ambient/architecture/index.md

```diff
@@ -33,9 +81,8 @@ However, depending on the destination's capabilities, different behavior will oc
 If the destination is also captured, or otherwise has Istio proxy capabilities (such as a sidecar), the request will be upgraded to an encrypted {{< gloss "HBONE" >}}HBONE tunnel{{< /gloss >}}.
 If the destination has a waypoint proxy, in addition to being upgraded to HBONE, the request will instead be forwarded to that waypoint.
 
-Note that in the case of a request to a `Service`, a specific endpoint will be selected to determine if it has a waypoint.
-However, if it *has* a waypoint, the request will be sent with a target destination of the `Service`, not the selected endpoint.
-This allows the waypoint to apply service-oriented policies to the traffic.
+Note that in the case of a request to a `Service`, if the service *has* a waypoint, the request will be sent to its waypoint to enforce service-oriented policies to the traffic.
```

Comment:
@ilrudie is it true that if a service has a waypoint, the traffic will be sent to the waypoint if and only if the waypoint is capable of handling `service` or `all` traffic? If so, should we clarify that here?

Similarly, below, w.r.t. Pod/Workload traffic.

Reply:
Hi @MorrisLaw I think what you said is true to my understanding. cc @howardjohn @stevenctl to weigh in as @ilrudie is going out of office for the next few days.

Reply:
My understanding is derived from the [waypoint capture doc](https://docs.google.com/document/d/1DAM_6SajCyC2hD8wBUraaZWZJqUVYPyemzw0-xpHmdY/edit):

```
The system assumes that any Gateway specified by istio.io/use-waypoint is capable of handling redirected traffic subject to this annotation and the binding specification. Since istio.io/use-waypoint plays a role in configuration generation too, it is useful to disable or more finely control redirection while allowing for testing of the generated config. Setting istio.io/waypoint-for: none we allow for this testing scenario.
```

Reply:
> if user tries to attach a waypoint that is svc-only to a workload that will be more or less a noop

good point

Reply:
Yep that is what i have observed, which makes sense. basically, if I label my pod use-waypoint: C, the pod won't use waypoint C until waypoint C is configured for workload (or all).

Reply:
Updated to make this clear, PTAL @MorrisLaw or @ilrudie  !!

Reply:
Lgtm

Reply:
Thanks, marking as resolved